h2/src/main/org/h2/store/fs/disk/FilePathDisk.java

```diff
@@ -55,23 +61,40 @@ public FilePathDisk getPath(String path) {
         return p;
     }
 
+    private FileSystem initFileSystem(URL url) throws Exception {
+        try {
+            return FileSystems.getFileSystem(url.toURI());
+        } catch (FileSystemNotFoundException e) {
+            Map<String, String> env = new HashMap<>();
+            env.put("create", "true");
+            return FileSystems.newFileSystem(url.toURI(), env);
+        }
+    }
+
     @Override
     public long size() {
         if (name.startsWith(CLASSPATH_PREFIX)) {
-            try {
-                String fileName = name.substring(CLASSPATH_PREFIX.length());
-                // Force absolute resolution in Class.getResource
-                if (!fileName.startsWith("/")) {
-                    fileName = "/" + fileName;
+            String path = this.name.substring("classpath:".length());
+            if (!path.startsWith("/")) {
+                path = "/" + path;
+            }
+            URL url = this.getClass().getResource(path);
+            if ("file".equals(url.getProtocol())) {
```

Comment:
A `NullPointerException` is now possible here, because `getResource()` may return `null`. Previously it was caught by outer `try-catch` block.

Reply:
Fixed.